src/Text/Markdown/SlamDown/Pretty.purs

```diff
@@ -121,28 +123,28 @@ prettyPrintTextBoxValue t =
     SD.Time prec (Identity def) → prettyPrintTime prec def
     SD.DateTime prec (Identity def) → prettyPrintDateTime prec def
 
-prettyPrintDate ∷ SD.DateValue → String
-prettyPrintDate { day, month, year } =
-  printIntPadded 4 year
+prettyPrintDate ∷ DT.Date → String
+prettyPrintDate d =
+  printIntPadded 4 (fromEnum $ DT.year d)
     <> "-"
-    <> printIntPadded 2 month
+    <> printIntPadded 2 (fromEnum $ DT.month d)
     <> "-"
-    <> printIntPadded 2 day
+    <> printIntPadded 2 (fromEnum $ DT.day d)
 
-prettyPrintTime ∷ SD.TimePrecision → SD.TimeValue → String
-prettyPrintTime prec { hours, minutes, seconds }=
-  printIntPadded 2 hours
+prettyPrintTime ∷ SD.TimePrecision → DT.Time → String
+prettyPrintTime prec t =
+  printIntPadded 2 (fromEnum $ DT.hour t)
     <> ":"
-    <> printIntPadded 2 minutes
+    <> printIntPadded 2 (fromEnum $ DT.minute t)
     <> case prec of
-        SD.Seconds -> ":" <> printIntPadded 2 (M.fromMaybe 0 seconds)
+        SD.Seconds -> ":" <> printIntPadded 2 (fromEnum $ DT.second t)
         _ -> ""
 
-prettyPrintDateTime ∷ SD.TimePrecision → SD.DateTimeValue → String
-prettyPrintDateTime prec { date, time } =
-  prettyPrintDate date
-    <> "T"
-    <> prettyPrintTime prec time
+prettyPrintDateTime ∷ SD.TimePrecision → DT.DateTime → String
+prettyPrintDateTime prec dt =
+  prettyPrintDate (DT.date dt)
+    <> " "
```

Comment:
After this it'll be much clearer... I think :wink:

- The markdown syntax will accept `YYYY-MM-DD HH:mm:SS` (or `YYYY-MM-DDTHH:mm:SS` for backwards compatibility). The pretty printer will always print `YYYY-MM-DD HH:mm:SS`, as that's the more human-readable form.
- The EJSON / SQL^2 timestamp syntax is exactly `YYYY-MM-DDTHH:mm:SSZ`. Both `T` and `Z` are required - that exact format is required by quasar, so our parser for it in EJSON is unforgiving also.

The markdown syntax is only used for parsing/printing markdown. Since the underlying type is now a `DateTime` value it means there's no hackery involved in formatting it again later, as the EJSON printer will take care of it and ensure it prints in the correct format there.